axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/factory/FixedAssetLineServiceFactory.java

```diff
@@ -51,4 +55,20 @@ public FixedAssetLineComputationService getFixedAssetComputationService(
             "There is no implementation for this typeSelect");
     }
   }
+
+  public FixedAssetLineService getFixedAssetService(int typeSelect) throws AxelorException {
+
+    switch (typeSelect) {
+      case FixedAssetLineRepository.TYPE_SELECT_ECONOMIC:
+        return Beans.get(FixedAssetLineEconomicServiceImpl.class);
+      case FixedAssetLineRepository.TYPE_SELECT_FISCAL:
+        return Beans.get(FixedAssetLineFiscalServiceImpl.class);
+      case FixedAssetLineRepository.TYPE_SELECT_IFRS:
+        return Beans.get(FixedAssetLineIfrsServiceImpl.class);
+      default:
+        throw new AxelorException(
+            TraceBackRepository.CATEGORY_NO_VALUE,
+            "There is no implementation for this typeSelect");
```

Comment:
Please create an error message as global constant in `AccountExceptionMessage` and add `i18n.get` on it. Also please make the message more functionally understandable.